python/rmm/_lib/logger.pyx

```diff
@@ -0,0 +1,301 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import warnings
+from enum import Enum, auto
+
+from libcpp cimport bool
+
+
+cdef extern from "spdlog/common.h" namespace "spdlog::level" nogil:
+    cdef enum level_enum:
+        trace = 0
+        debug = 1
+        info = 2
+        warn = 3
+        err = 4
+        critical = 5
+        off = 6
```

Comment:
In order to make this directly usable in Python you should be able to do
```suggestion
    cpdef enum logging_level "level_enum":
        trace
        debug
        info
        warn
        err
        critical
        off
```

There's three main changes here:
1. Using `cpdef` instead of `cdef` automatically creates a Python wrapper around the enum so that you don't have to.
2. The `logging_level "level_enum"` basically says "this object is actually called level_enum in C, but in Cython I want to call it logging_level". The one small downside is that in the `cdef cppclass` block below you'll have to change `level_enum` to `logging_level` in all of the function parameters, but then everything else should work.
3. No need to specify the values, you can inherit them from the C enum.

Reply:
I think this will require using the lower-case enum names from the C++ enum rather than the pythonic uppercase ones. Is that OK?

Reply:
Good call, we should be able to alias those as well using the same quoting trick. Something like `TRACE "spdlog::level::A"` should work (these are unscoped enums so I assume accessing directly from the `level` namespace will work).

Reply:
Done.

Reply:
Thanks for the guidance here.